src/renderers/dom/client/wrappers/ReactDOMInput.js

```diff
@@ -70,11 +70,11 @@ var ReactDOMInput = {
     var value = LinkedValueUtils.getValue(props);
     var checked = LinkedValueUtils.getChecked(props);
 
-    var nativeProps = assign({
+    var nativeProps = DisabledInputUtils.assignWithNativeProps({
       // Make sure we set .type before any other properties (setting .value
       // before .type means .value is lost in IE11 and below)
       type: undefined,
-    }, DisabledInputUtils.getNativeProps(inst, props), {
```

Comment:
I'm keeping this `{ type: undefined }` default for now, but is this relying on key insertion order? If not, could we replace this with something later like:

```javascript
// Make sure we set .type, otherwise .value is lost in IE11 and below
nativeProps.type = nativeProps.type == null ? undefined: nativeProps
```